test/classes/delete-free/owned-init.chpl

```diff
@@ -0,0 +1,18 @@
+use OwnedObject;
+
+class C {
+  var x: int;
+
+  proc matches(other) {
+    if this == other then
+      writeln("same!");
+    else
+      writeln("different!");
+  }
+}
+
+proc main() {
+  const a = [i in 1..3] new Owned(new C(i));
```

Comment:
Pretty sure this is the same problem as init2; just within the compiler generated promoted routine.